### PR TITLE
merge the common content in demo values file

### PR DESCRIPTION
**Pull request by linsun**
fixes: #12270

### CODE REVIEW

**Comment by linsun** [comment]
thanks @howardjohn agreed on the source and released file out of sync concern, but that is what we wanted to have all common values in 1 file...

**Comment by linsun** [comment]
@mandarjog since it is a common file, it should not have any conflict, else it should not be put in the common file.

as of today, there is no conflict.  I did validate it is okay to have multiple global blobs like:

```
global:
  controlPlaneSecurityEnabled: false

  mtls:
    # Default setting for service-to-service mtls. Can be set explicitly using
    # destination rules or service annotations.
    enabled: false
# This is used to generate minimal demo mode. It is included from demo and demo-auth values.
# It is shipped with the release, used for bookinfo or quick installation of istio.
# Includes components used in the demo, defaults to alpha3 rules.
global:
  proxy:
    accessLogFile: "/dev/stdout"
    resources:
      requests:
        cpu: 10m
        memory: 40Mi

  disablePolicyChecks: false

  sidecarInjectorWebhook:
    enabled: true
```

**Comment by linsun** [comment]
BTW, you have the same handle conflict issue when specify it as helm param via ```--values file1.yaml --values file2.yaml```

**Comment by mandarjog** [comment]
@linsun multiple helm value args have clearly defined merge semantics. 

In that case for manual merging you should have the common file prepended not appended.

File: release/create_release_archives.sh [hunk 1]
```diff
@@ -139,6 +139,13 @@ find install/tools -type f -exec "${CP}" --parents {} "${COMMON_FILES_DIR}" \;
 find tools -type f -not -name "githubContrib*" -not -name ".*" -exec "${CP}" --parents {} "${COMMON_FILES_DIR}" \;
 popd
 
+ls -l  "${COMMON_FILES_DIR}/install/kubernetes/"
+ls -l  "${COMMON_FILES_DIR}/install/kubernetes/helm/istio"
+
+# merge values-istio-demo-common.yaml into values-istio-demo yaml files
+cat"${COMMON_FILES_DIR}/install/kubernetes/helm/istio/values-istio-demo-common.yaml" >> "${COMMON_FILES_DIR}/install/kubernetes/helm/istio/values-istio-demo.yaml"
```
Comment (howardjohn) [inline release/create_release_archives.sh]: missing a space after `cat`

Comment (linsun) [reply release/create_release_archives.sh]: yes corrected, thanks.

@mandarjog I also changed the file to insert common file first.

**Comment by linsun** [comment]
@duderino @wenchenglu can we get this minor fix but improves user experience in 1.1?

**Pull request merged**